src/GitHub.TeamFoundation.14/Base/TeamExplorerServiceHolder.cs

```diff
@@ -14,23 +14,17 @@ public class TeamExplorerServiceHolder : ITeamExplorerServiceHolder
     {
         IServiceProvider serviceProvider;
 
-        /// <summary>
-        /// This class relies on IVSGitExt that provides information when VS switches repositories.
-        /// </summary>
-        /// <param name="gitExt">Used for monitoring the active repository.</param>
-        [ImportingConstructor]
-        TeamExplorerServiceHolder(ITeamExplorerContext teamExplorerContext) :
-            this(teamExplorerContext, ThreadHelper.JoinableTaskContext)
-        {
-        }
-
         /// <summary>
         /// This constructor can be used for unit testing.
         /// </summary>
         /// <param name="gitService">Used for monitoring the active repository.</param>
         /// <param name="joinableTaskContext">Used for switching to the Main thread.</param>
-        public TeamExplorerServiceHolder(ITeamExplorerContext teamExplorerContext, JoinableTaskContext joinableTaskContext)
+        [ImportingConstructor]
+        public TeamExplorerServiceHolder(ITeamExplorerContext teamExplorerContext,
+            [Import(AllowDefault = true)] JoinableTaskContext joinableTaskContext)
         {
+            joinableTaskContext = joinableTaskContext ?? ThreadHelper.JoinableTaskContext;
+
             JoinableTaskCollection = joinableTaskContext.CreateCollection();
             JoinableTaskCollection.DisplayName = nameof(TeamExplorerServiceHolder);
             JoinableTaskFactory = joinableTaskContext.CreateFactory(JoinableTaskCollection);
```

Comment:
I know you didn't change it, but can you explain to me what is going on with this collection?

Reply:
It's creating a collection that tasks created by `JoinableTaskFactory` will be placed into. It's then possible to wait for all tasks in the collection to complete before disposing of the service when Visual Studio shuts down. This pattern was copied from another service.

In this case the service isn't disposable and the `JoinableTaskCollection` is never used. It could be simplified to simply store the `JoinableTaskContext` and use the default collection (which is associated with `JoinableTaskContext.Factory`).

I think it would make sense to be consistent and include this simplification as part of this PR.

Reply:
OK, cool.